### PR TITLE
Align to Grayskull and Decrease jupyter_core floor

### PR DESCRIPTION
**Pull request by rxm7706**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Align to GraySkull, leaving entry points
Decreases  jupyter_core floor to version available in conda
https://github.com/elyra-ai/elyra/pull/3071

### CODE REVIEW

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,28 +33,27 @@ requirements:
     - entrypoints >=0.3
     - jinja2 >=3
     - jsonschema >=3.2.0,<4.0
-    - jupyter-resource-usage >=0.5.1
     - jupyter_client >=6.1.7
+# https://github.com/elyra-ai/elyra/issues/3069    
     - jupyter_core >=4.11.2,<5
```
Comment (kevin-bates) [inline recipe/meta.yaml:38]: Let's remove the comment and the cap.  There's no reason to penalize all Python versions and 3.7 doesn't contain a `jupyter_core == 5` release anyway.
```suggestion 
    - jupyter_core >=4.11.2
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -33,28 +33,27 @@ requirements:
     - entrypoints >=0.3
     - jinja2 >=3
     - jsonschema >=3.2.0,<4.0
```
Comment (kevin-bates) [inline recipe/meta.yaml:35]: Several adjustments were made to dependencies and caps in https://github.com/elyra-ai/elyra/pull/3076, so we should apply them here.

File: recipe/meta.yaml (hunk 1 above)
Comment (kevin-bates) [reply recipe/meta.yaml:38]: See also https://github.com/elyra-ai/elyra/pull/3071 as it applies a conditional cap to `nbclient` relative only to Python 3.7 (in addition to removing `jupyter_core`'s cap).  Hopefully there's a way to apply the same kind of conditional behavior here.

**Review by kevin-bates — COMMENTED** [review]
Thank you for making this better!  We recently (this past week) landed a couple of pull requests that adjust other dependencies.  It would be good to apply those changes here as well.

File: recipe/meta.yaml (hunk 1 above)
Comment (rxm7706) [reply recipe/meta.yaml]: I don't think we will need to add this one, it just wont resolve - I believe to do any python version specific conditions - we would need to make the entire elyra package move away from noarch

**Comment by rxm7706** [comment]
@kevin-bates thank you. 

All the changes have been made - but that causes the pip check to fail. 

Both the pins are correct in the meta.yaml and match the recent changes. 
    "jupyter_core>=4.11.0",  # Do not increase floor past 4.11.1 until Python 3.7 EOL
    "jupyter-events>=0.6.2",

This should work once the pypi package has been updated with the next release from main. 

Current Error from build for referance
```
import: 'elyra'
+ pip check
elyra 3.14.1 has requirement jupyter-core<5,>=4.11.2, but you have jupyter-core 5.1.3.
elyra 3.14.1 has requirement jupyter-events<0.5.0, but you have jupyter-events 0.6.3.
Tests failed for elyra-3.14.1-pyhd8ed1ab_1.conda - moving package to /home/conda/feedstock_root/build_artifacts/broken
```

**Comment by kevin-bates** [comment]
> This should work once the pypi package has been updated with the next release from main.

Oh right, sorry about that.  How should we proceed with this PR?  Can we wait until Elyra 3.14.2 or 3.15?

**Comment by rxm7706** [comment]
I've reverted the changes that caused pip check to fail, we can include them in the next version - this build should work for Py3.7 users. 
> Oh right, sorry about that. How should we proceed with this PR? Can we wait until Elyra 3.14.2 or 3.15?

**Review by kevin-bates — APPROVED** [review]
Looks good - thank you @rxm7706.

**Pull request merged**